examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "56c9e4d7-03fc-4a30-81f8-83d63bab170a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "from traitlets import Int\n",
+    "\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    my_width = Int(23).tag(sync=True)\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton({ my_width }) {\n",
+    "      return (\n",
+    "        <button\n",
+    "          style={{\n",
+    "            position: \"relative\",\n",
+    "            width: my_width,\n",
+    "            height: 30,\n",
+    "          }}\n",
+    "        >\n",
+    "          {\" \"}\n",
+    "          Width of {my_width} px{\" \"}\n",
+    "        </button>\n",
+    "      );\n",
+    "    }\"\"\"\n",
+    "MyExampleWidget(my_width=300)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4dba603f-8052-4d8d-845d-d28029e17c35",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "MyExampleWidget(my_width=600)\n",
+    "#MyExampleWidget(my_width=\"Hello\") # <- this will throw an error, as my_width expects an integer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93f2c43e-5a25-4787-a7a4-11f71a7ddbd1",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# you can also create an instance of a class\n",
+    "w = MyExampleWidget(my_width=600)\n",
+    "w"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "10d5d10c-2254-4be3-ac2b-66b851120b03",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# and display this instance multiple times\n",
+    "w"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dc012a29-d639-48bd-b18f-21d687063e0d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "w.my_width = 400 # see how this will change the button width"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9dcc8417-b470-4981-aa77-4382e81882af",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "w.my_width = 200 # and one more time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8c2956f1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w.my_width # and get current parameter like this"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1463b79a",
+   "metadata": {},
+   "source": [
+    "Next, we will **add a counter** to this widget.\n",
+    "\n",
+    "There are two options:\n",
+    "🌱 Firstly, we can define a `my_count` traitlet and for the type we choose an Int traitlet.  \n",
+    "Note that for for every trait <name> there is a `<name> + on_<name>` pair on the react/frontend side.  \n",
```

Comment:
"Note that for for every trait <name> there is a `<name> + on_<name>` pair of values available for props of the component on the react side"

the `on_<name>` value is a setter function equivalent to what you get from react's `useState`.

That is still awkwardly worded.

Draw the parallel between val/on_val to `useState`'s `val` `setVal` that react devs will be used to.


Also this might be a good time to mention which naming format ipyreact leans towards.

There is python snake_case, and `on_` then there is is react camelCase and `setVal` style.  I don't care which one we recommend, but recommending one or the other will make things easier.

Generally when I have been coding my react inside of jupyter stuff, I use python naming at the boundaries, but for internal components that are never exposed, I use react style naming.  Reason being, if I want help from a more experienced frontend dev, I want them to see code that looks like what they're used to.

Reply:
yeah, the js part should be camelCase, but I kept the `on_value` pythonic because indeed, it is a reminder that this interfaces with the python side. So I think we agree here!

Reply:
Also, from this part it's not clear that `on_<value>` can be used to change the value from the js side. Also, maybe `set_<value>` would be a better name.. ?